source/app/service-providers/commands/file-new.ts

```diff
@@ -113,30 +134,28 @@ export default class FileNew extends ZettlrCommand {
         throw new Error('Could not create file: Filename was not valid')
       }
 
-      if (!hasMdOrCodeExt(filename)) {
-        // There's no valid file extension given. We have to add one. By default
-        // we assume Markdown, but let ourselves be guided by the given type.
+      // Check if the file already has an extension, and don't add the default extension if it does
+      if (!path.extname(filename)) {
```

Comment:
Please undo the changes in the if/switch construction here.

First, `if(!path.extname(filename))` does not pass the linter, you'd need to explicitly compare it to a string w/o implicit boolean conversion. Second, this construction should've already worked. Or is there something problematic with how it works right now? Did I overlook something?